### ISSUE TITLE
[BUG] Editor not applying customisations made before installation

### ISSUE THREAD

**Issue by sysdotini** [CriticalFlaw/TF2HUD.Editor#57]
# Enabling "Code Pro Fonts" doesn't enable Code Pro Fonts
Title is pretty self-explanatory. This seems to be an editor issue as manually appending `_pro` to `clientscheme.res` still works as intended.

## Reproduction
1) Remove any old huds from tf/custom
2) Launch the latest TF2 HUD Editor
3) Select FlawHUD and your custom folder
4) Check "Code Pro Fonts" under Customisations
5) Save changes & launch TF2. Code Pro Fonts will not be applied, and you will have the defaults.

If you view `flawhud/resource/clientscheme.res`, you will see the file just hasn't changed. Manually editing it to include "pro" still works as intended.

## Tested On
I've had this issue for a while, so I've tested it on different builds of my OS at this point. Additionally, I've tried launching it with elevated permissions just as a test, and the behaviour is the same.

Haven't had time to test from the main branch instead of the latest release.

- Windows 10 19043 (21H1)
- Windows 10 21390 (21H2 Dev)
- TF2Hud Editor v2.1

**Comment by cooolbros** [comment]
I cannot replicate this with the latest release version of the editor. Can you try installing FlawHUD from https://github.com/CriticalFlaw/flawhud ?

**Comment by sysdotini** [comment]
FlawHUD's Code Pro fonts work completely fine if I manually enable them.

I am only experiencing this issue with the latest release version of the editor.

If it helps, it seems it removes the comment in clientscheme.res instructions users how to enable them manually - but doesn't append _pro to it.

**Comment by CriticalFlaw** [comment]
Could you please try to reproduce this issue with the 2.1.1 preview build? It includes improved logging, so if you can reproduce the issue we can read the logs to get a better idea of where the problem is happening. Thanks!

https://github.com/CriticalFlaw/TF2HUD.Editor/releases/tag/2.1.1

**Comment by sysdotini** [comment]
The issue still happens on v2.1.1. However, I've seemed to narrow down the issue using the latest dev build.

If you choose to install a hud and tick a customization **before you install** it, it will not save any of your customizations. However, if you press "apply changes" after you install it, it seems to write said changes.

I *assume* this is mostly-intended behavior as it doesn't write customizations on download - but it leads to me (as an end-user) wondering *why* it didn't work on initial install. Maybe push customizations as a task after the download?

- As "Code Pro Fonts" *seems* ticked by default (correct me if I'm wrong!), I would  have assumed that it would enable them - even on a fresh install.

Also, I cleaned up AppData/Local & any temp files the application may have created to make sure it wasn't getting my previous customizations. I may have missed something though, so do correct me if I'm wrong!

## Logs
I've uploaded the logs to Pastebin to avoid flooding the thread.

1) [Initial Install][1]
2) [Applying Changes][2]

## TL;DR
- Customizations **do not save** when ticked upon a **fresh** HUD install
- Customizations **do save** when changes are "applied" to an **existing** install

Apologies if this was 100% intentional behavior. It makes sense to me as a developer, but it would be *much* better UX IMHO if it was "fixed" - or at least an indication that you must tick customizations after the initial install.

[1]: https://pastebin.com/3N4e4pZE
[2]: https://pastebin.com/Eb4DJr6m

**Comment by CriticalFlaw** [comment]
It looks like you're right. After installation we save the user settings file but we don't actually apply them. @cooolbros can correct me if I'm wrong, but I don't think this is intentional. This should be corrected to apply customizations after an installation, unless there's a good reason not to. I'll wait for his response.

Thank you for investigating this!

**Comment by CriticalFlaw** [comment]
Removing the flawhud tag because this is an editor issue present with all the huds.

**Comment by cooolbros** [comment]
Added in a3969e4e9ae68e152c4c0529d29691b68fa4cc4a. Thanks for reporting this issue

**Comment by CriticalFlaw** [comment]
Closing this issue as it's been fixed and will be included in version 2.2
